### ISSUE TITLE
Comments: comment specific view UI

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#19773]
As part of #18321 and a follow up to #19635, let's make sure we can see the details of a specific comment. Note that this does not include comment param actions. (From say an email moderation link.)

Right now we can see a comment render when we visit `/comment/{ siteFragment }/{ commentId }`

### Outstanding Items To Follow Up On:
* redirect to the site or post view when deleting a comment permanently here
* add feedback when replying to a comment
* add edit action to the component when available

@copons also notes:

* The page header reuses CommentListHeader which was built with the Post View in mind.
So, right now, the Comment View has an header that contains the post title, date, and a "View Post" link. It doesn't make sense.
cc @drw158 we oughta change this, likely with a new lean and clean header component in /my-sites/comment.

* Somehow the possibility to externally change a comment status made an oldie but goldie issue emerge.
Try to mark a comment as spam via mail link or direct URL (e.g. comments/123/example.org/456?action=spam): we land on a comment; this comment has a white background, and no labels.
In other words: we don't have any clues at all to know that our comment is spam (but the same happens for trash) and, by extension, that our action was completed successfully.
(cc @drw158)

* PS to the previous point: do we need to display a notice to confirm the action?

* PPS: damn, the "undoable" Comment actions don't work in this view because they depends on the CommentList.updateLastUndo function.
We should check for updateLastUndo === undefined before firing it in CommentActions, since we wouldn't need it in Comment View.

**Comment by drw158** [comment]
@Copons I'm finally getting to this. Wasn't quite sure what your comments meant but I get it now!

I noticed some things in the edit comment screen (or "history" screen) wp-admin:

- Permalink is only visible on approved comments
- This screen is not accessible for trashed comments (his might be a problem for us if moderating via email)
- Editing the comment text is available for all comments but trashed.

Here's an idea for spam comments:

- Headercake now reads "Edit comment" with the date of the comment
- Date is removed in the comment card
- Comment has edit button (should we just add the edit button for spam comments everywhere?)
- Time to bring back the pill label for spam?
- View post button is removed
- No permalink

![artboard copy 26](https://user-images.githubusercontent.com/618551/32963390-690b90f6-cb95-11e7-96e4-2949e838c6ae.png)

**Comment by Copons** [comment]
@gwwar I think this issue contains a couple of quick wins that should be tackled sooner rather than later.
I could open self-contained issues for them.

cc @drw158 for UX suggestions

---

The permalink is shown for all comments, regardless of their status.
This works for approved and pending comments, since they both show up on the site (pending are visible for moderators); though it doesn't make sense for spam and trash, since the `#` anchor wouldn't be anchored to anything, and so the link would just be a (rather confusing) link to the post.

**Let's hide the permalink for spam and trash comments**.

---

Can you guess what status is this comment?

<img width="785" alt="screen shot 2018-02-02 at 11 24 36" src="https://user-images.githubusercontent.com/2070010/35730903-b3f35934-080b-11e8-9fda-6b432caadc11.png">

Ok, maybe our team could actually guess it based on which actions are displayed (it's a trashed comment).

Though, normal users would have a hard time figuring out, because both spam and trash look almost identical to approved.

We do use the "Pending" pill label (which is unnecessary-ish, since pending comments have a yellow background anyway): **let's reuse it for spam and trash comments in single view**.

---

When we land on this view from outside (e.g. from notifications, or from edit link, etc.) we usually have the `?action=edit` param which forces the edit mode.

Though, the edit mode by itself (meaning: loading into it, as opposed to naturally getting into it with a user-driven flow of clicking buttons and navigating around) it's quite unclear.
For example, we have the "Save" and "Cancel" buttons. We can disregard if the former is unclear (e.g. "where will I be redirected after saving my changes?"), but what about the latter?
In fact, the "Cancel" button just disable the edit mode, and replaces the "Save"/"Cancel" buttons with the status actions.

This is not really an expected behaviour, especially if the user is not familiar with Comments Management (very likely, if they get here from the outside).
So, I'm proposing 2 action items here:

- **Make the status actions always visible when in edit mode, in single comment view**: this is because maybe a user got here to, for example, mark a comment as spam ("Edit" is a generic term, it doesn't necessarily mean "edit the content").

- **Change the "Cancel" button label (again, in single comment view)** into something that might convey the whole "exit edit mode" meaning.

**Comment by drw158** [comment]
> Let's hide the permalink for spam and trash comments.
> Allow for editing spam and trash comments.
> We do use the "Pending" pill label (which is unnecessary-ish, since pending comments have a yellow background anyway): let's reuse it for spam and trash comments in single view.

Makes senses

> Make the status actions always visible when in edit mode, in single comment view:

I think if you click on "Edit" somewhere externally, then maybe we should just link to the single comment view, not the edit mode.

**Comment by Copons** [comment]
> I think if you click on "Edit" somewhere externally, then maybe we should just link to the single comment view, not the edit mode. The user will see the Edit button and go from there.

That's what I wanted to do for the edit button in the notifications.
And yet:

> I noticed that when we land in Calypso, it's not really obvious what a user should do (they still have to click Edit again). Did we not set up a query param at some point that loaded the view with the Edit section open?..
https://github.com/Automattic/notifications-panel/pull/222#issuecomment-362040067

So, I agree with you, but I can also see the point that we're asking users to click on the Edit button twice (in the notifs/email/frontend, and then in the comments management) to get to the edit mode.

At the same time, I can't see any real reason to not show the actions while in edit mode AND in single view.

Let me clarify:
In LIST view, the Edit button toggles between [view mode + status actions] and [edit mode].
In SINGLE view, the Edit button would toggle between [view mode] and [edit mode], while the status actions remains always visible.

Or, again in SINGLE view, **we could load with the actions visible only when we come from outside Calypso** (which is when we've got the `?action=edit` query param).
With this option (which at this point I think it's the best), we're giving basically the same experience as wp-admin: both Edit mode and status actions.
Then, if the user toggles stuff manually (e.g. exit edit mode, and then enter it again), the status actions visibility reverts back to the standard behaviour of being hidden in Edit mode, and visible in View mode.

⬆️This is much more complicated to explain than it is to code, though!

**Comment by gwwar** [comment]
> When we land on this view from outside (e.g. from notifications, or from edit link, etc.) we usually have the ?action=edit param which forces the edit mode.

This was changed recently since folks were using the front end link to moderate not edit. see https://github.com/Automattic/wp-calypso/issues/21931
